test/JDBC/expected/BABEL-CASE_EXPR-vu-verify.out

```diff
@@ -13014,3 +13014,1522 @@ GO
+END AS Result
+GO
+~~START~~
+sql_variant
+1.23
+~~END~~
+
+
+-- Test Case 41: Numeric branch with time
+select case 1
+   when 1 then cast(1.23 as numeric(20,2))
+   when 2 then cast(1900-01-01 as datetime)
+END
+go
+~~START~~
+datetime
+1900-01-02 05:31:12.0
+~~END~~
+
+
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('1900-01-01' AS DATE)
+END AS Result
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: CASE/WHEN could not convert type date to numeric)~~
+
+
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('12:34:56' AS TIME)
+END AS Result
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: CASE/WHEN could not convert type time without time zone to numeric)~~
+
+
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('1900-01-01 12:34:56.1234567' AS DATETIME2(7))
+END AS Result
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: CASE/WHEN could not convert type datetime2 to numeric)~~
+
+
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('1900-01-01 12:34:56 +01:00' AS DATETIMEOFFSET)
+END AS Result
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: CASE/WHEN could not convert type datetimeoffset to numeric)~~
+
+
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('1900-01-01 12:34:00' AS SMALLDATETIME)
+END AS Result
+GO
+~~START~~
+smalldatetime
+1900-01-02 05:31:00.0
+~~END~~
+
+
+SELECT CASE 2
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST('1900-01-01' AS DATE)
+   WHEN 3 THEN CAST('12:34:56' AS TIME)
+   WHEN 4 THEN CAST('1900-01-01 12:34:56.1234567' AS DATETIME2(7))
+   WHEN 5 THEN CAST('1900-01-01 12:34:56 +01:00' AS DATETIMEOFFSET)
+   WHEN 6 THEN CAST('1900-01-01 12:34:00' AS SMALLDATETIME)
+END AS Result
+GO
+~~START~~
+smalldatetime
+1900-01-01 00:00:00.0
+~~END~~
+
+
+-- Test Case 42: NUMERIC and TINYINT
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST(255 AS TINYINT)
+END AS Result
+GO
+~~START~~
+numeric
+1.23
+~~END~~
+
+
+-- Test Case 43: NUMERIC and REAL
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST(3.14 AS REAL)
+END AS Result
+GO
+~~START~~
+real
+1.23
+~~END~~
+
+
+-- Test Case 44: NUMERIC and NCHAR
+SELECT CASE 1
+   WHEN 1 THEN CAST(1.23 AS NUMERIC(20,2))
+   WHEN 2 THEN CAST(N'A' AS NCHAR(1))
+END AS Result
+GO
+~~START~~
+numeric
+1.23
+~~END~~
+
+
+-- Create UDTs
+CREATE TYPE NumericType_BABEL_5341 FROM NUMERIC(20,2);
+GO
+
+CREATE TYPE DecimalType_BABEL_5341 FROM DECIMAL(18,4);
+GO
+
+CREATE TYPE BigNumericType_BABEL_5341 FROM NUMERIC(38,10);
+GO
+
+CREATE TYPE SmallDecimalType_BABEL_5341 FROM DECIMAL(5,2);
+GO
+
+-- Test Case 45: Basic CASE with all UDTs
+SELECT CASE 4
+   WHEN 1 THEN CAST(12345.6789 AS NumericType_BABEL_5341)
+   WHEN 2 THEN CAST(98765.4321 AS DecimalType_BABEL_5341)
+   WHEN 3 THEN CAST(1234567.8901234567890 AS BigNumericType_BABEL_5341)
```

Comment:
kindly use 18 digits before decimal for value with type `BigNumericType_BABEL_5341`, so result precision can also be verified.